test/i18n_viz_test.rb

```diff
@@ -51,3 +51,36 @@ class I18nVizJavascriptIntegrationTest < ActionDispatch::IntegrationTest
     assert page.has_css?("#i18n_viz_tooltip")
   end
 end
+
+class I18nVizDynamicUrlTest < ActionDispatch::IntegrationTest
+  setup do
+    Capybara.current_driver = Capybara.javascript_driver # :selenium by default
+
+    # monkey patch external_tool_url
```

Comment:
Well I tried but after Rails initializes the middleware it freezes the array. I tried deleting or swapping the current instance of I18nViz middleware with another one, which has the lambda configured but without luck. Maybe it is due to Rails specifics, I don't know, but this is the only way I managed to get this test working.